Feature/blog-authors

zysim opened the pull request:
This branch is to solve #12. Stuff I've done include:
- Every blog post made so far should print out the author at the bottom now
- `data.authors` created in `_config.yml` which enables my previous bullet point, plus extra info
- Author's avatar will be replaced by the first letter of their name if we don't have an image

zysim committed:
Add packages in Gemfile
------
Added what hopefully should be all required packages to develop this
project.

zysim committed:
Add authors list to _config.yml
------
Added the list of authors to the config YAML file. Each blog post will
now display who wrote that post, along with some other tidbits. We're
like a real blog now

zysim committed:
Add backup to logo in author blurb
------
In the author blurb that is written at the bottom of every blog post,
it now displays the first letter of the author's name if we don't have
an avatar for them yet. Also added an SCSS file to style that.

zysim committed:
Change Camil's author front matter entry
------
Hey dude changed your name to lower case to maintain consistency
in data.authors in
_config.yml

takluyver commented on _config.yml:
```diff
@@ -63,3 +63,77 @@ exclude:
+      name: Thomas
+      avatar:
+      web:
+      bio: Thomas seeks world domination
```
Wow, looks like there's a load of competition for this world domination thing ;-)

zysim replied:
Oh wow yeah I totally forgot to remove those default texts haha feel free to edit your own though

takluyver commented:
This sounds OK to me - I assume you've tested building it and it works as described.

If you've got time to investigate, maybe you could try to fix the netlify builds - it looks like they're running `jekyll serve` when they should be running `jekyll build`. I've no idea where that's controlled from, though.

zysim commented:
Aye I'll look into the netlify build when I can. @Nanoseb Ok for me to merge?

Nanoseb commented:
Arg, really sorry for the delay, I saw it but then forgot about it...
Concerning the bio ;) how visible is the text? is it on every blog post? if so maybe leave it empty for now, and every one can add its own.

Commit:
Merge branch 'gh-pages' into feature/blog-authors

Commit:
Add article-specific author bio
------
A new front matter element, author_bio can be placed in an article,
which acts as an override for the author's default bio that is set in
_config.yml. The code to render default bios is currently commented out
till we create actually decent bios.

zysim commented:
Updating this PR with a merge from gh-pages, and a new commit, [ba148b](https://github.com/Maritime-Robotics-Student-Society/blog/pull/13/commits/ba148b36c6cf4bdd303caf85c9ea5a4db8872cb0). This new commit allows an optional, article-specific author bio, christened `author_bio` that is placed in the front matter for every article. If it is set, it will be displayed in the bio at the bottom of the article page. This _is_ also meant to default to the bios set in `_config.yml`, but the code for that is currently commented out, until we create decent default bios for everyone. Hence, if `author_bio` isn't set, it'll just display the author's name and nothing else.
